qcfractal/web_handlers.py

```diff
@@ -335,17 +335,46 @@ class ProcedureHandler(APIHandler):
     _required_auth = "read"
     _logging_param_counts = {"id"}
 
-    def get(self):
+    def get(self, query_type='get'):
 
-        body_model, response_model = rest_model("procedure", "get")
+        body_model, response_model = rest_model("procedure", query_type)
         body = self.parse_bodymodel(body_model)
 
         try:
-            ret = self.storage.get_procedures(**{**body.data.dict(), **body.meta.dict()})
+            if query_type == 'get':
+                ret = self.storage.get_procedures(**{**body.data.dict(), **body.meta.dict()})
+            else:  # all other queries, like 'best_opt_results'
+                ret = self.storage.query('procedure', query_type, **{**body.data.dict(), **body.meta.dict()})
         except KeyError as e:
             raise tornado.web.HTTPError(status_code=401, reason=str(e))
 
         response = response_model(**ret)
 
         self.logger.info("GET: Procedures - {} pulls.".format(len(response.data)))
         self.write(response)
+
+class OptimizationHandler(APIHandler):
```

Comment:
This looks duplicated from above? We could also consider a `CustomQueryHandler` generic.

`localhost:7777/custom/optimization/`

Reply:
This won't be quite REST, since you have to have the resource first. Also, in reality, I think all methods should be under the same object, the custom_queries are not actually custom from the user perspective, it has a different implementation in the backend, that's it.

So, I was thinking that all custom queries will be under their main REST resource. So if we add a new query for optimizaton, it will automatically go through `optimization/my_query_type/`. Same for tasks, etc.

Reply:
Sure I like `procedure/...`. My main comment here is that this class seems extraneous. The tornado handler points to a `/optimization/...` resource at the moment.